core/src/main/scala/scalaz/IList.scala

```diff
@@ -475,9 +475,10 @@ sealed abstract class IList[A] extends Product with Serializable {
 
 }
 
-final case object INil extends IList[Nothing] {
-  def apply[A](): IList[A] = this.asInstanceOf[IList[A]]
-  def unapply[A](e: IList[A]): Boolean = this eq e
+sealed abstract case class INil[A] private() extends IList[A]
```

Comment:
argh, this means we can't use a wart to detect uses of non-`final` case class. This will incur additional "allow the wart here" markers.

Reply:
That wart should probably change then.